### PR TITLE
[5.3] Add method Builder::lastInsertId()

### PR DESCRIPTION
**Pull request by vlakoff**
Use case:

``` php
$query->insert($bulk);

// before (long, not DB-agnostic)
$idFirst = $query->getConnection()->getPdo()->lastInsertId();
// after (zonda)
$idFirst = $query->lastInsertId();

$idLast = $idFirst + count($bulk) - 1;
```

PostgreSQL is a bit tricky, see http://stackoverflow.com/questions/2944297/postgresql-function-for-last-inserted-id#2944481. Current `processInsertGetId()` uses option 3, my added `processLastInsertId()` uses option 1.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Could you not just use `insertGetId` from the beginning?

**Comment by vlakoff** [comment]
Currently `insertGetId` doesn't support inserting several rows at once.

It could be modified to support this, however the lastInsertedId corresponds *to the first inserted row*.
I thought it's better to not complicate `insertGetId` and rather add a specific method.

**Comment by vlakoff** [comment]
And yes, I really have to insert by batches.
Otherwise I'm running a SQL query for each inserted row, and thousands of SQL queries is *a bit* slow :trollface:

**Comment by taylorotwell** [comment]
You have tested this on Postgres?

File: src/Illuminate/Database/Query/Processors/SqlServerProcessor.php [hunk 1]
```diff
@@ -9,20 +9,16 @@
...
-     * @param  array   $values
      * @param  string  $sequence
      * @return int
      */
-    public function processInsertGetId(Builder $query, $sql, $values, $sequence = null)
+    public function processLastInsertId(Builder $query, $sequence = null)
     {
         $connection = $query->getConnection();
 
-        $connection->insert($sql, $values);
```
Comment (taylorotwell) [inline src/Illuminate/Database/Query/Processors/SqlServerProcessor.php]: Why was this removed?

Comment (vlakoff) [inline src/Illuminate/Database/Query/Processors/SqlServerProcessor.php]: Look at the single commit "Add method Builder::lastInsertId()".
Then `processInsertGetId` is identical to the parent one, and I remove it in the following commit "Same as parent".

**Comment by vlakoff** [comment]
The postgresql code works, though it returns the id of the *last* inserted row from the batch…

**Comment by vlakoff** [comment]
So, a better solution would be to:
* make `insertGetId` accept several rows to insert (in fact this seems pretty easy to do)
* and consistently returns id of last inserted row with all drivers.

**Comment by vlakoff** [comment]
Closing this PR because of the mysql discrepancy, and I guess the modifications suggested in above comment wouldn't be accepted.
